galley/pkg/runtime/resource/schema.go

```diff
@@ -52,28 +52,28 @@ func newSchemaBuilder(messageTypeFn messageTypeFn) *SchemaBuilder {
 }
 
 // Register a proto into the schema.
-func (b *SchemaBuilder) Register(typeURL string) Info {
-	if _, found := b.schema.byURL[typeURL]; found {
-		panic(fmt.Sprintf("schema.Register: Proto type is registered multiple times: %q", typeURL))
+func (b *SchemaBuilder) Register(rawCollection, typeURL string) Info {
```

Comment:
I thought we replace typeURL with collection, why do we keep them both?

Reply:
A collection has a corresponding schema (e.g. typeURL) defined by the proto. This mapping is defined in galley/tools/gen-meta/metadata.yaml and is registered at runtime here. The typeURL is used elsewhere to allocate new protobuf instances of the correct type.